river/main.zig

```diff
@@ -139,12 +139,17 @@ fn defaultInitPath() !?[:0]const u8 {
         }
     };
 
-    os.accessZ(path, os.X_OK) catch |err| {
+    os.accessZ(path, os.R_OK) catch |err| {
         std.log.err("failed to run init executable {s}: {s}", .{ path, @errorName(err) });
         util.gpa.free(path);
         return null;
     };
 
+    os.accessZ(path, os.X_OK) catch {
+        std.log.err("failed to run init executable {s}: the file is not executable", .{path});
+        os.exit(1);
+    };
+
```

Comment:
Instead of always doing two syscalls, could we only do the `R_OK` check in the error path?